sequences/README.md

```diff
@@ -0,0 +1,282 @@
+# ...
+```
+
+```sh
+$ curl -si -H "Content-Type: application/json" -d '[12, "foo"]' localhost:8080/connect
+HTTP/1.1 502 Bad Gateway
+Content-Type: application/json
+Couper-Error: endpoint error
+...
+
+{
+  "error": {
+    "id":      "c71kuad916bht5pkuqb0",
+    "message": "endpoint error",
+    "path":    "/connect",
+    "status":  502
+  }
+}
+```
+
+Now we see only one entry in the `couper_backend` log (for `/add`).
+
+We also see an error in the `couper_access` log:
+
+```json
+{
+  "error_type": "unexpected_status",
+  "level": "error",
+  "message": "endpoint error",
+  "type": "couper_access"
+}
+```
+
+Let's add an `error_handler` to handle the error:
+
+```hcl
+...
+      request {
+        ...
+        expected_status = [200]
+      }
+      error_handler "unexpected_status" {
+        response {
+          status = 500
+          json_body = {
+            error = "upstream error"
+            error_description = "an upstream service responded with an unexpected status code"
+          }
+        }
+      }
+# ...
+```
+
+```sh
+$ curl -si -H "Content-Type: application/json" -d '[12, "foo"]' localhost:8080/connect
+HTTP/1.1 500 Internal Server Error
+Content-Type: application/json
+...
+
+{"error":"upstream error","error_description":"an upstream service responded with an unexpected status code"}
+```
+
+And we can log some additionaly information about the requests in the case of an error:
+
+```hcl
+# ...
+      error_handler "unexpected_status" {
+        response {
+          ...
+        }
+        custom_log_fields = {
+          add = {
+            message = backend_responses.add.json_body.error.message
+            status = backend_responses.add.json_body.error.status
+          }
+          default = {
+            message = backend_responses.default.json_body.error.message
+            status = backend_responses.default.json_body.error.status
+          }
+        }
+      }
+# ...
+```
+
+This adds a new field to the log message:
+
+```json
+{
+  "custom": {
+    "add": {
+      "message": "expression evaluation error",
+      "status": 500
+    }
+  },
+  "error_type": "unexpected_status"
+}
+```
+
+showing that the proxy `"add"` responded with a status code `500` and the error message `"expression evaluation error"`.
+
+If we change the `request` to
+
+```hcl
+# ...
+      request {
+        url = "http://math:8081/multiply"
+        json_body = [ backend_responses.add.json_body.result, "bar" ]  # ← "bar" instead of 4
+      }
+# ...
+```
+and send a "proper" array, the `custom` field in the log message now shows that proxy `"add"` produced a "proper" result, while request `"default"` has an error:
+
+```json
+{
+  "custom": {
+    "add": {},
+    "default": {
+      "message": "expression evaluation error",
+      "status": 500
+    }
+  },
+  "error_type": "unexpected_status"
+}
+```
+
+---
+
+Another use case for a sequence is the provisioning of an access token prior to a request that must be authorized.
+
```

Comment:
add the docker-compose cmd. and also add one or two curl examples to show it in action.